source/adapters/level_zero/kernel.cpp

```diff
@@ -229,6 +233,12 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   if (IndirectAccessTrackingEnabled)
     Queue->KernelsToBeSubmitted.push_back(Kernel);
 
+  if (!Queue->AppendedEvents) {
+    Queue->AppendedEvents = new ze_event_handle_t[Queue->totalCmdListCount - 1];
+  }
+  Queue->AppendedEvents[Queue->CmdListEnqueued] = ZeEvent;
```

Comment:
This event is host visible, you need to create a device only scoped event for each event and then set the HostVisibleEvent=ZeEvent